packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/DiagramPanel.tsx

```diff
@@ -50,18 +59,32 @@ export const DiagramPanel = ({
   const handleShare = () => setState((prevState) => ({ ...prevState, dialogOpen: 'Share' }));
   const handleCloseDialog = () => setState((prevState) => ({ ...prevState, dialogOpen: null }));
 
-  const reactFlowInstance = useReactFlow();
   const { fadeDiagramElements } = useFadeDiagramElements();
   const { hideDiagramElements } = useHideDiagramElements();
 
   const onUnfadeAll = () => fadeDiagramElements([...getAllElementsIds()], false);
   const onUnhideAll = () => hideDiagramElements([...getAllElementsIds()], false);
 
+  const handleExport = () => {
+    const imageWidth: number = window.screen.width;
+    const imageHeight: number = window.screen.height;
+    const nodesBounds: Rect = getRectOfNodes(reactFlow.getNodes());
+    const transform: Transform = getTransformForBounds(nodesBounds, imageWidth, imageHeight, 0.5, 2);
+
+    toSvg(document.querySelector<HTMLElement>('.react-flow__viewport')!, {
+      backgroundColor: '#ffffff',
+      width: imageWidth,
+      height: imageHeight,
+      style: {
+        width: imageWidth.toString(),
+        height: imageHeight.toString(),
+        transform: `translate(${transform[0]}px, ${transform[1]}px) scale(${transform[2]})`,
+      },
+    }).then(downloadImage);
```

Comment:
the '!' should not be used like this.
It is the same as doing `optional.get()` without doing `if(optional.isPresent())` before in Java.
 
Prefer something like
```typescript
const viewport: HTMLElement | null = document.querySelector<HTMLElement>('.react-flow__viewport');
if (viewport) {
  toSvg(...)
}
```

Reply:
Done